includes/data-stores/class-wc-admin-reports-variations-data-store.php

```diff
@@ -91,11 +100,10 @@ protected function get_sql_query_params( $query_args ) {
 			$sql_query_params['where_clause'] .= " AND {$order_product_lookup_table}.variation_id IN ({$allowed_variations_str})";
 		}
 
-		if ( is_array( $query_args['order_status'] ) && count( $query_args['order_status'] ) > 0 ) {
-			$statuses = array_map( array( $this, 'normalize_order_status' ), $query_args['order_status'] );
-
-			$sql_query_params['from_clause']  .= " JOIN {$wpdb->prefix}posts ON {$order_product_lookup_table}.order_id = {$wpdb->prefix}posts.ID";
-			$sql_query_params['where_clause'] .= " AND {$wpdb->prefix}posts.post_status IN ( '" . implode( "','", $statuses ) . "' ) ";
+		$order_status_filter = $this->get_status_subquery( $query_args );
+		if ( $order_status_filter ) {
+			$sql_query_params['from_clause']  .= " JOIN {$wpdb->prefix}wc_order_stats ON {$order_product_lookup_table}.order_id = {$wpdb->prefix}wc_order_stats.order_id";
+			$sql_query_params['where_clause'] .= " AND ( {$order_status_filter} )";
```

Comment:
Is there an opportunity to DRY this up? Perhaps a base "joins on wc_order_stats" class that handles the `from_clause`, `where_clause`, `normalize_order_by()`, and the `orders_count` column?

Reply:
Hmm, I'm not sure another class would be the best solution here.  Report data stores already extend from `WC_Admin_Reports_Data_Store`, but quite a few also override the `normalize_order_by()` and other methods.

The bigger issue here is that there are about 3-4 different patterns being used in these data stores  to append query parameters which makes it difficult to create a universal class that does not result in overriding in most data stores.  For example, here are a few different variations just for the "where" clause:
```php
$sql_query_params['where_clause'] .= " AND ( {$order_status_filter} )";

$where_subclause = implode( " $operator ", array_filter( array( $where_subclause, $order_status_filter ) ) );

$products_where_clause .= " AND ( {$order_status_filter} )";

$where_filters[] = $order_status_filter;

$sql_query_params['where_clause'] .= " AND ( {$order_status_filter} )";
$sql_query_params['where_clause'] .= ' AND variation_id > 0';
```

It seems like we could make a call outside of and adjacent to the `get_sql_query_params()` that might work for most data stores, but that would also be semantically incorrect and we'd be introducing more patterns that may come back to haunt us later.

I do agree that it would be nice if this were dryer, but it seems like a fairly large refactor that might be outside the scope of this PR.  If we decided to do something like that, it might be worth breaking into bite-size chunks (like just refactoring `get_sql_query_params()` to use the same patterns) to create more manageable PRs.